### ISSUE TITLE
negation operator is needed to make json schema working correctly 

### ISSUE THREAD

**Issue by Saibo-creator** [epfl-dlab/transformers-CFG#57]
It seems currently the json schema constrain is not working correctly yet 

by loading the example student grammar

```python
 # Load grammar
 with open("examples/grammars/custom_json_grammars/grammars/student.ebnf", "r") as file:
      grammar_str = file.read()
```

And run a simple generation we get :

```terminal
{ "name": "\\"\\"", "age": 20, "courses": [ "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"", "\\"\\"",']
```

The origin of the problem is that the grammar converted from the json schema, i.e. `examples/grammars/custom_json_grammars/grammars/student.ebnf` contains a usage of negation operator `^` at line 4, i.e. `char ::= [^"\\]`

Since we haven't yet added the support for negation operator , this simply doesn't work and only allows to generate escaped backslashes.   #21 

A quick fix is to replace it with `char ::= [A-z]`

And we get 
```terminal
{ "name": "John", "age": 20, "courses": [ "Math", "English" ] }
```

**Comment by Saibo-creator** [comment]
As a follow up work of #21 , we should now be able to support json schema. Negation was the only missing operator.  See here for more details of [how to use json schema](https://github.com/epfl-dlab/transformers-CFG/blob/main/examples%2Fgrammars%2Fcustom_json_grammars%2FREADME.md)

**Comment by Saibo-creator** [comment]
solved in #21
